apps/src/lib/node/ledger/ethereum_node/mod.rs

```diff
@@ -28,15 +28,27 @@ pub enum Error {
 
 pub type Result<T> = std::result::Result<T, Error>;
 
-/// Monitor the Ethereum fullnode. If it stops or an abort
-/// signal is sent, the subprocess is halted.
+/// Monitor the Ethereum fullnode subprocess, returning only once it has
+/// stopped running. If a signal is sent on `abort_recv`, the fullnode
+/// subprocess will be killed.
 pub async fn monitor(
     mut node: eth_fullnode::EthereumNode,
     abort_recv: Receiver<Sender<()>>,
-) -> Result<()> {
+) {
```

Comment:
The send half of the `abort_recv` channel here belongs to `AbortableSpawner`, and is what is actually used to shut down the Ethereum fullnode if needed. Made this fn return `()` instead of `Result<()>` since it was always returning `Ok(())` or panicking anyway.